### ISSUE TITLE
Empty collection literals type is undetermined inside spread ternary.

### ISSUE THREAD

**Issue by im-aIex** [dotnet/roslyn#69949]
### Description

The collection literal spec states that:
_Spreading an empty literal is permitted to be elided. For example:_
```csharp
bool b = ...
List<int> l = [x, y, .. b ? [1, 2, 3] : []];
```
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-12.0/collection-expressions#empty-collection-literal

However, that exact example gives the error:
> error CS0173: Type of conditional expression cannot be determined because there is no implicit conversion between 'collection literals' and 'collection literals'

### Reproduction Steps

[SharpLab example](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGED2AbbEDGMAlpgHYAyRMECU2yANACYgDUAPgAIBMAjALAAoIUVIwABAA9xAXnG8G4gJ6zx3ANxChwTDnHBVMBHAibBomAG0AuuOyrLkxUsUA6V/vEB+cZYVrFAGZbEF9razMhTl4ATgAKaNcAFQhJGFcAKWQyTOzSAGUaIjoiAC8aV0KEYuwyiDjsAEpGsyA==)

### Expected behavior

The type is determined without a syntax error.

### Actual behavior

The type is undetermined and it gives an error.

### Known Workarounds

Explicitly cast the empty collection literal or the entire ternary operation.

### Configuration

.NET 8 rc1

**Comment by jaredpar** [comment]
Believe nested collection expression inference won't be until .NET 9. Will let @cston  confirm

**Comment by CyrusNajmabadi** [comment]
It's currently a decision we intend to bring to LDM to discuss.  Haven' tmade a decision on it yet.  But def on the potential 'cut list'.

**Comment by im-aIex** [comment]
Intellisense also suggests making changes which result in the broken code above, which is how I initially discovered it.

**Comment by CyrusNajmabadi** [comment]
@im-aIex yes, the IDE features are currently written as per the lang specification at thsi time.  If we aren't able to ship all those lang-spec features in the compiler, we'll change the IDE features.

**Comment by jaredpar** [comment]
@cston, @CyrusNajmabadi can we get confirmation on the cut / no cut for this for C# 12?

**Comment by cston** [comment]
Nested collection expressions won't make C#12.

**Comment by CyrusNajmabadi** [comment]
This was cut.  THere are proposals open currently to support this for the future.
